services/accumulo-utils/src/main/java/datawave/marking/MarkingFunctions.java

```diff
@@ -68,18 +73,29 @@ public Exception(Throwable cause) {
         }
     }
     
-    class NoOp implements MarkingFunctions {
+    class Default implements MarkingFunctions {
         public static final String COLUMN_VISIBILITY = "columnVisibility";
         
         @Override
         public ColumnVisibility combine(Collection<ColumnVisibility> expressions) {
-            return expressions.isEmpty() ? new ColumnVisibility() : expressions.iterator().next();
+            
+            // filter out any empty expressions, then flatten each one (to de-dupe) and concatenate with '&'
+            // flatten the final combined ColumnVisibility and use that to make the ColumnVisibility to return
+            return new ColumnVisibility(new ColumnVisibility(expressions.stream().filter(viz -> viz.flatten().length > 0)
+                            .map(viz -> "(" + new String(viz.flatten(), UTF_8) + ")").collect(Collectors.joining("&")).getBytes(UTF_8)).flatten());
         }
         
         @Override
         @SafeVarargs
         public final Map<String,String> combine(Map<String,String>... markings) {
-            return markings.length == 0 ? new HashMap<>() : markings[0];
+            Map<String,String> combinedMap = new HashMap<>();
+            Stream.of(markings).flatMap(map -> map.entrySet().stream().filter(entry -> !Strings.isNullOrEmpty(entry.getValue()))).forEach(entry -> {
```

Comment:
My guess is you probably want to operate only on the COLUMN_VISIBILITY entry in the map, if there is one.